spec/spec.md

```diff
@@ -0,0 +1,174 @@
+# Verona Language Spec
+
+## Lexical Elements
+
+### Comments
+
+### Identifiers and Symbols
+
+- `_` as "don't care".
+
+### Strings
+
+### Constants
+
+- Boolean
+- Various integers and floats
+
+> How to talk about integer literals not having a type yet?
+
+### Lambdas
+
+### Object Literals
+
+- Open issues: code reuse, possible conflict with "real" `new`.
+
+### Expression Termination
+
+- Blank lines
+- New lines
+- Indentation
+- Trailing lambdas
+
+## Program Structure
+
+- It's all classes: modules are classes
+- Fields
+- Functions
+- "Left-hand" functions
+- Nesting
+- Default values
+- Visibility and encapsulation
+- Generics: what needs to be discussed here as opposed to in `Types`?
+  - Treat it as template-like, discuss type checking later
+- Do we need to talk about type lists here?
+
+## Expression Structure
+
+### Bindings
+
+- Parameters
+- Let
+- Var
+- References
+
+### Grouping with Parentheses
+
+### Conditionals
+
+### Static Accessors: Double Colons
+
+### Dots
+
+- Selectors
+- Reverse application
+
+### Universal Call Syntax
+
+- Object-like vs. operator-like local names
+  - Type names are `create` sugar
```

Comment:
do no issues arise about order of intiailization of objects?

Reply:
Comments from discussion
* Static initialisers in other languages, and when things happen. 
* Immutable static data initialisation, probably not a problem here.
* Initialisation order is complex in some languages.
* `create` as a factory method, and allocates after it has found all the bits. So create can fail, make sure we allow failure returns from `create`.